### DISCUSSION TITLE
Cannot connect to mongodb with .env variable

### DISCUSSION THREAD

**Discussion by ssayanm** [strapi/strapi#6251] [Legacy Topics]
Hi ! I am trying to connect a MongoDB URI to the strapi backend. I am able to connect with the main URL but when I created a .env.development variable, I am not able to connect to the database. I have kept the .env file at the root folder of my backend

```
{
  "defaultConnection": "default",
  "connections": {
    "default": {
      "connector": "mongoose",
      "settings": {
        "uri": "${process.env.DATABASE_URI || '127.0.0.1'}"
      },
      "options": {
        "ssl": true
      }
    }
  }
}
```

**Comment by lauriejim** [comment]
Hello! I think it could be come from the parser, do you have any " ' or other character in you `process.env.DATABASE_URI` that could cause this issue.

I just tried w/ a MongoDB Atlas uri and it works fine.

`DB_URI="mongodb://admin:strapi@cluster0-shard-00-00-avwic.gcp.mongodb.net:27017,cluster0-shard-00-01-avwic.gcp.mongodb.net:27017,cluster0-shard-00-02-avwic.gcp.mongodb.net:27017/test?ssl=true&replicaSet=Cluster0-shard-0&authSource=admin&retryWrites=true&w=majority" yarn start`

**Comment by ssayanm** [comment]
Hi Jim .. I manged to connect it...thank you so much for your message.  I watched all your videos in youtube, I am learning Strapi and I want to move from wordpress to (strapi + gatsby).

**Reply by derrickmehaffy** [reply]
Native .env support is already in the rc.x releases and will be in the Stable release which is coming very soon (ETA was end of this month I believe)

**Comment by bamne123** [comment]
In this situation, Space should be handled by the code. Trim function should called for all the Environment variables.
